src/tests/daos_perf.c

```diff
@@ -1398,10 +1403,25 @@ main(int argc, char **argv)
 	}
 
 	if (ts_mode == TS_MODE_VOS) {
+		if (ts_ctx.tsc_mpi_size > 1 &&
+		    (access("/etc/daos_nvme.conf", F_OK) != -1)) {
+			fprintf(stderr,
+				"no support: multi-proc vos_perf with NVMe\n");
+			return -1;
+		}
 		ts_ctx.tsc_cred_nr = -1; /* VOS can only support sync mode */
-		if (strlen(ts_pmem_file) == 0)
-			strcpy(ts_pmem_file, "/mnt/daos/vos_perf.pmem");
-
+		if (strlen(ts_pmem_file) == 0) {
+			int len = snprintf(NULL, 0, "/mnt/daos/vos_perf%d.pmem",
+					   ts_ctx.tsc_mpi_rank) + 1;
+			snprintf(ts_pmem_file, len, "/mnt/daos/vos_perf%d.pmem",
+				 ts_ctx.tsc_mpi_rank);
+		} else {
+			char id[4];
+			int  len = snprintf(NULL, 0, "%d",
+					    ts_ctx.tsc_mpi_rank);
+			snprintf(id, len, "%d", ts_ctx.tsc_mpi_rank);
+			strcat(ts_pmem_file, id);
```

Comment:
```suggestion
			strncat(ts_pmem_file, id, sizeof(ts_pmem_file));
```

Reply:
good point strcat should be changed to strncat but we should use strlen(ts_pmem_file) instead of sizeof

Reply:
You should specify how many bytes can be concatenated maximum. So, correct size should be
sizeof(ts_pmem_file) - strlen(ts_pmem_file)